consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProvider.java

```diff
@@ -75,15 +80,15 @@ public ForkingValidatorProvider(
   }
 
   private Collection<Address> getValidators(
-      final long block, final Function<ValidatorProvider, Collection<Address>> getValidators) {
-    final BftForkSpec<QbftConfigOptions> forkSpec = forksSchedule.getFork(block);
-    final ValidatorProvider validatorProvider = resolveValidatorProvider(block);
+      final ValidatorProvider validatorProvider,
+      final BftForkSpec<QbftConfigOptions> forkSpec,
+      final Function<ValidatorProvider, Collection<Address>> getValidators) {
 
     // when moving to a block validator the first block needs to be initialised or created with
     // the previous block state otherwise we would have no validators
-    if (!forkSpec.getConfigOptions().isValidatorContractMode()) {
-      if (block > 0 && block == forkSpec.getBlock()) {
-        final long prevBlockNumber = block - 1L;
+    if (forkSpec.getConfigOptions().isValidatorBlockHeaderMode()) {
+      if (forkSpec.getBlock() > 0) {
```

Comment:
Removing `block == forkSpec.getBlock()` changes the behaviour and breaks a unit test, need to figure out if changing the meaning of the unit test is the right fix.